### PR TITLE
Publish JARs for trigger service and OAuth 2.0 middleware

### PR DESCRIPTION
**Pull request by aherrmann-da**
- Adds the oauth2-middleware logback config, I overlooked it in https://github.com/digital-asset/daml/pull/8611
- Adds steps to publish trigger service and OAuth 2.0 middleware JARs on GitHub.

Thanks @garyverhaegen-da for helping me with this!

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: release/artifacts.yaml [hunk 1]
```diff
@@ -179,3 +179,7 @@
   type: jar-scala
 - target: //triggers/runner:trigger-runner-lib
   type: jar-scala
+- target: //triggers/service:trigger-service
```
Comment (cocreature) [inline release/artifacts.yaml:182]: I wouldn’t publish them to maven. These are executables not supported libraries.

File: release/util.bzl [hunk 2]
```diff
@@ -73,6 +73,7 @@ def sdk_tarball(name, version):
           cp $(location //daml-assistant/daml-sdk:sdk_deploy.jar) $$OUT/daml-sdk/daml-sdk.jar
           cp -L $(location //ledger-service/http-json:src/main/resources/logback.xml) $$OUT/daml-sdk/json-api-logback.xml
           cp -L $(location //triggers/service:release/trigger-service-logback.xml) $$OUT/daml-sdk/
+          cp -L $(location //triggers/service/auth:release/oauth2-middleware-logback.xml) $$OUT/daml-sdk/
```
Comment (cocreature) [inline release/util.bzl:76]: why did we not get a test failure despite this?

**Review by cocreature — APPROVED** [review]
Awesome, thanks!

**Review by garyverhaegen-da — APPROVED** [review]
LGTM if the goal is to publish via GH (and not Artifactory).

File: release/util.bzl (hunk 2 above)
Comment (aherrmann-da) [reply release/util.bzl:76]: Passing a non-existing path to the `--logback-config` flag on `daml-helper run-jar` seems to be silently ignored.

**Comment by aherrmann-da** [comment]
I've verified that the JARs were uploaded to GH on https://github.com/digital-asset/daml/pull/8625, that the shasums match, and that the JARs can be executed with `java -jar` and the `/livez` endpoint of the trigger service and oauth2 middleware return the expected response.

**Pull request merged**